### ISSUE TITLE
Questionable CTDOXY from I6S

### ISSUE THREAD

**Issue by mvkovatch3** [SIO-ODF/ctdcal#10]
Oxygen processing seems bad for I6. My initial guess is the incorrect weighting where surface bottles are weighted as most accurate and deep bottles are least.

Examples of bad casts shown below:

![image](https://user-images.githubusercontent.com/57368341/83910687-d9e96d80-a71f-11ea-8485-d7709d893ca4.png) ![image](https://user-images.githubusercontent.com/57368341/83910780-00a7a400-a720-11ea-958f-f4f2305a2429.png)

Not all are completely bad:
![image](https://user-images.githubusercontent.com/57368341/83910835-13ba7400-a720-11ea-94c3-3b8e287350a7.png) ![image](https://user-images.githubusercontent.com/57368341/83910851-1a48eb80-a720-11ea-8e6b-71b1dbf4ed32.png)

There's also a couple that are bad due to multiple oxygen sensor issues from the cruise overall:
![image](https://user-images.githubusercontent.com/57368341/83910919-3482c980-a720-11ea-9985-53c4eebcd0c4.png)

> Hi Susan,
> 
> Are you back to the office yet? How is it?
> 
> I'm writing regarding the I06S CTD data. Kats wrote me yesterday to point to a few stations with negative CTDOXY values, but with FLAG = 2 (stations are 5 to 8). Looking quickly at the other stations, it looks like stations 9, 11 and 34 have some bad values too, but they're marked "questionable".
> 
> Not super urgent, but could you please double check the quality flag of the dataset?
> 
> Many thanks!!
> isa
> 
> 
> --
> Isa Rosso, Ph.D. 
> Scripps Institution of Oceanography 
> University of California, San Diego 
> 
> phone: +1 415 570 1042
> skype: isazar
> https://irosso.scrippsprofiles.ucsd.edu/

**Comment by DocOtak** [comment]
This is awesome (as a bug report)!

**Comment by mvkovatch3** [comment]
Cruise data is only partially backed up to server. @asx- is helping track down the rest of the files so they can be properly backed up and also reprocessed

**Comment by DocOtak** [comment]
See "Other Files" https://cchdo.ucsd.edu/cruise/325020190403

**Comment by asx-** [comment]
> Cruise data is only partially backed up to server. @asx- is helping track down the rest of the files so they can be properly backed up and also reprocessed

Data happens to be on servers and timestamped to May 2019 - computer glitch didn't copy all the data down? No upload of backup data to the servers.

**Comment by mvkovatch3** [comment]
> > Cruise data is only partially backed up to server. @asx- is helping track down the rest of the files so they can be properly backed up and also reprocessed
> 
> Data happens to be on servers and timestamped to May 2019 - computer glitch didn't copy all the data down? No upload of backup data to the servers.

There seems to be some kind of permissions error when using rsync, e.g.
`rsync: send_files failed to open "/share/odf/ODF-Cruises/325020190403_I06S/ctd/raw/00502.XMLCON": Permission denied (13)`

It seems like my file copying only grabbed files set to `rwxr-xr-x`

Snippit of `ls -lah` in `/share/odf/ODF-Cruises/325020190403_I06S/ctd/raw`:
![image](https://user-images.githubusercontent.com/57368341/84057645-d0a11080-a96c-11ea-8ad0-f1146b92388a.png)

**Comment by DocOtak** [comment]
Might need to kick this up to a JIRA ticket for one of the STS folks with "proper" credentials to look at...

**Comment by asx-** [comment]
Confirmed permissions are not resetable using my account. Something is wrong with the permissions, and a JIRA ticket will need to be made.

**Comment by mvkovatch3** [comment]
[JIRA ticket](https://somts.ucsd.edu/support/browse/ODF-78) created, waiting for response from CR. Work on this issue is paused until file permissions are remedied

**Comment by mvkovatch3** [comment]
A month to the day, data looks ready for final review and re-submission.

### Fixes employed
#### 1) Remove unrealistic values
- interpolate over unrealistic pressure data (e.g. deeper than cast depth; >6500 dbar)
- interpolate over salinity data beyond observed range + fudge factor (33.5 - 35.75)
- interpolate over unrealistic oxygen data (e.g. <0 or >1000 umol/kg)

#### 2) Despike
- despike T/C/O using rolling 24-point stdev filter; interpolate over points >20 times average stdev (repeated 3x)

_[Note for (1) and (2), interpolation is only performed over gaps up to 24 points long.]_

#### 3) Update oxygen fitting routine
- fix density matching following PMEL routine

### Updated data examples
Same casts which were previously identified as bad:
![image](https://user-images.githubusercontent.com/57368341/87103487-09513580-c20a-11ea-8515-ed8c04e5c24d.png) ![image](https://user-images.githubusercontent.com/57368341/87103490-0c4c2600-c20a-11ea-9081-be14782212fc.png)

Same casts which were previously identified as partially bad:
![image](https://user-images.githubusercontent.com/57368341/87103505-19691500-c20a-11ea-907a-b70d9aa88dde.png) ![image](https://user-images.githubusercontent.com/57368341/87103509-1b32d880-c20a-11ea-907f-a082e0c6319d.png)

Previously unsalvageable data:
![image](https://user-images.githubusercontent.com/57368341/87103518-2554d700-c20a-11ea-9e48-4e9bb03f8b6f.png)

**Comment by DocOtak** [comment]
I imagine there will need to be updates to the ctd oxygen values bottle file as well? I could probably handle putting that stuff in the data file on CCHDO... should probably have a "subject matter expert" (Jim) look at the data before pulling that publish button (I wish it was a button).

**Comment by mvkovatch3** [comment]
Yep, I'm having Jim take a look and assuming we get his stamp of approval we can deal with publishing.
